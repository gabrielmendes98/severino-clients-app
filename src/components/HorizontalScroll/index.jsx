import React from 'react';
import PropTypes from 'prop-types';
import { ScrollView, View } from 'react-native';
import styles from './style';

const HorizontalScroll = ({ children }) => (
  <ScrollView
    horizontal
    showsHorizontalScrollIndicator={false}
    style={styles.scrollView}
  >
    <View style={styles.view}>{children}</View>
  </ScrollView>
);

HorizontalScroll.propTypes = {
  children: PropTypes.node.isRequired,
};

export default HorizontalScroll;
