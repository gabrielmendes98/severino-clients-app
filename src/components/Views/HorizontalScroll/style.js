import { StyleSheet } from 'react-native';
import { SAFE_SPACING } from 'common/constants';
import theme from 'common/styles/theme';

const styles = StyleSheet.create({
  scrollView: {
    marginHorizontal: theme.spacing(-SAFE_SPACING),
  },
  view: {
    flexDirection: 'row',
    marginLeft: theme.spacing(SAFE_SPACING),
  },
});

export default styles;
