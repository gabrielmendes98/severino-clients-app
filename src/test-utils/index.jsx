/* eslint-disable react/prop-types */
import React from 'react';
import { render } from '@testing-library/react-native';
import MainProvider from 'common/contexts/Main';

const Wrapper = ({ children }) => <MainProvider>{children}</MainProvider>;

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
