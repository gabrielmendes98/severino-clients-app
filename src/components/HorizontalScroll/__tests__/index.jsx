import React from 'react';
import { View } from 'react-native';
import { rawRender } from 'test-utils';
import HorizontalScroll from '../index';

it('should renders correctly', () => {
  const { toJSON } = rawRender(
    <HorizontalScroll>
      <View></View>
    </HorizontalScroll>,
  );

  expect(toJSON()).toMatchSnapshot();
});
