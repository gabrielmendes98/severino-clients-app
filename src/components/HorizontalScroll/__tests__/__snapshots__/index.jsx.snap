// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correctly 1`] = `
<RCTScrollView
  horizontal={true}
  showsHorizontalScrollIndicator={false}
  style={
    Object {
      "marginHorizontal": -12,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginLeft": 12,
        }
      }
    >
      <View />
    </View>
  </View>
</RCTScrollView>
`;
